### PR TITLE
Add a logging message consumer wrapper that conforms to the LSP inspector debug tool

### PR DESCRIPTION
**Pull request by matthewmichihara**
Fixes https://github.com/eclipse/lsp4j/issues/328.

Still work to do here, but sharing to get early feedback. Does this look like a reasonable direction?
For context, here is some of the parsing code that the inspector uses: https://github.com/microsoft/language-server-protocol-inspector/blob/master/lsp-inspector/src/logParser/rawLogParser.ts

https://microsoft.github.io/language-server-protocol/inspector/

image

### CODE REVIEW

File: org.eclipse.lsp4j.jsonrpc/src/main/java/org/eclipse/lsp4j/jsonrpc/LoggingMessageConsumerWrapper.java [hunk 1]
```diff
@@ -0,0 +1,132 @@
+package org.eclipse.lsp4j.jsonrpc;
```
Comment (spoenemann) [inline org.eclipse.lsp4j.jsonrpc/src/main/java/org/eclipse/lsp4j/jsonrpc/LoggingMessageConsumerWrapper.java]: Add a copyright header comment (see other files).

File: org.eclipse.lsp4j.jsonrpc/src/main/java/org/eclipse/lsp4j/jsonrpc/LoggingMessageConsumerWrapper.java [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+
+import java.io.PrintWriter;
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Function;
+
+public class LoggingMessageConsumerWrapper implements Function<MessageConsumer, MessageConsumer> {
```
Comment (spoenemann) [inline org.eclipse.lsp4j.jsonrpc/src/main/java/org/eclipse/lsp4j/jsonrpc/LoggingMessageConsumerWrapper.java]: Rename to `TracingMessageConsumerWrapper` (or simply `MessageTracer`).

File: org.eclipse.lsp4j.jsonrpc/src/main/java/org/eclipse/lsp4j/jsonrpc/LoggingMessageConsumerWrapper.java [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+                        NotificationMessage notificationMessage = (NotificationMessage) message;
+                        action = "Sending notification";
+                        method = notificationMessage.getMethod();
+                        traceLine = String.format("[Trace - %s] %s '%s'", date, action, method);
+
+                        Object params = notificationMessage.getParams();
+                        String paramsJson = MessageJsonHandler.toString(params);
+                        logLine = String.format("Params: %s", paramsJson);
+                    } else {
+                        throw new IllegalStateException("Unhandled message: " + message);
```
Comment (spoenemann) [inline org.eclipse.lsp4j.jsonrpc/src/main/java/org/eclipse/lsp4j/jsonrpc/LoggingMessageConsumerWrapper.java]: I would rather print an error to the output `printWriter` than throw an exception here. Same for other throws below.

**Comment by spoenemann** [comment]
You need to [sign the Eclipse Foundation ECA](https://www.eclipse.org/legal/ECA.php) and sign-off your commit (`git commit -s`) in order to pass the contribution requirements.

**Comment by matthewmichihara** [comment]
@spoenemann ready for re-review, ptal when you get a chance.

**Comment by spoenemann** [comment]
ok to test

**Comment by matthewmichihara** [comment]
Thanks for the merge! Out of curiosity, any thoughts on when the next release will be?

**Comment by KamasamaK** [comment]
@matthewmichihara https://projects.eclipse.org/projects/technology.lsp4j

**Pull request merged**